django/db/models/constraints.py

```diff
@@ -26,6 +27,15 @@ def clone(self):
         _, args, kwargs = self.deconstruct()
         return self.__class__(*args, **kwargs)
 
+    def set_name_with_dict(self, model):
+        try:
+            self.name = self.name % {
+                'app_label': model._meta.app_label.lower(),
+                'class': self.__class__.__name__.lower()
+            }
+        except FieldDoesNotExist:
```

Comment:
I was wondering what raises `FieldDoesNotExist` here?

Reply:
to fix the key errors but I checked and that doesn't work. I will change with `KeyError`. the purpose is to catch like these samples `name='%(app_label1)...'`

Reply:
I think it is better not to suppress `KeyError`, it means that user is doing something wrong by giving `app_label1` there and needs to know about it.

Reply:
you're absolutely right.